blackjax/adaptation/mclmc_adaptation.py

```diff
@@ -298,6 +302,310 @@ def step(state, key):
     return adaptation_L
 
 
+Lratio_lowerbound = 0.0
+Lratio_upperbound = 2.0
+
+
+def adjusted_mclmc_find_L_and_step_size(
```

Comment:
The code added here has quite a bit of duplication with the non adjusted version:
```
mclmc_find_L_and_step_size
make_L_step_size_adaptation
make_adaptation_L
```

I suggest you create a new file, that way when we are comparing the implementation and trying to abstract the 2 it is a bit easier. WDYT?

Reply:
OK, I split it out. There is similar code, but the details of the respective functions differ in important ways. It's true that with effort, one could probably abstract the code a little, but I'm not yet convinced that the gain would be huge. At a later date, we're planning to do some refactoring of the tuning (https://github.com/blackjax-devs/blackjax/issues/677) and that might be a good point to revisit this.